lua/rustaceanvim/commands/diagnostic.lua

```diff
@@ -391,4 +411,56 @@ function M.render_diagnostic_current_line()
   render_ansi_code_diagnostic(rendered_diagnostic)
 end
 
+---@class rustaceanvim.diagnostic.RelatedInfo
+---@field location lsp.Location
+---@field message string
+
+function M.related_diagnostics()
+  local ra = require('rustaceanvim.rust_analyzer')
+  local clients = ra.get_active_rustaceanvim_clients(0)
+  if #clients == 0 then
+    return
+  end
+  ---@type lsp.Location[]
+  local locations = vim
+    .iter(get_diagnostics_at_cursor())
+    ---@param diagnostic vim.Diagnostic
+    :map(function(diagnostic)
+      return vim.tbl_get(diagnostic, 'user_data', 'lsp', 'relatedInformation')
+    end)
+    :flatten()
+    ---@param related_info rustaceanvim.diagnostic.RelatedInfo
+    :map(function(related_info)
+      return related_info.location
+    end)
+    :totable()
+  if #locations == 0 then
+    vim.notify('No related diagnostics found.', vim.log.levels.INFO)
+    return
+  end
+  local quickfix_entries = vim.lsp.util.locations_to_items(locations, clients[1].offset_encoding)
+  if #quickfix_entries == 1 then
+    local item = quickfix_entries[1]
+    ---@diagnostic disable-next-line: undefined-field
+    local b = item.bufnr or vim.fn.bufadd(item.filename)
+    -- Save position in jumplist
+    vim.cmd.normal { "m'", bang = true }
+    -- Push a new item into tagstack
+    local tagstack = { { tagname = vim.fn.expand('<cword>'), from = vim.fn.getpos('.') } }
+    local current_window_id = vim.api.nvim_get_current_win()
+    vim.fn.settagstack(vim.fn.win_getid(current_window_id), { items = tagstack }, 't')
+    vim.bo[b].buflisted = true
+    local window_id = vim.fn.win_findbuf(b)[1] or current_window_id
+    vim.api.nvim_win_set_buf(window_id, b)
+    vim.api.nvim_win_set_cursor(window_id, { item.lnum, item.col - 1 })
+    vim._with({ win = window_id }, function()
+      -- Open folds under the cursor
+      vim.cmd.normal { 'zv', bang = true }
+    end)
+  else
+    vim.fn.setqflist({}, ' ', { title = 'related diagnostics', items = quickfix_entries })
+    vim.cmd.botright('copen')
```

Comment:
@mrcjkb This command doesn't work for me on Neovim v0.10.2:

```
E5108: Error executing lua [string ":lua"]:1: Vim:E476: Invalid command
stack traceback:
        [C]: in function 'botright'
        [string ":lua"]:1: in main chunk
```

Reply:
Oops 😬 

- https://github.com/mrcjkb/rustaceanvim/pull/605